server.js

```diff
@@ -1,14 +1,22 @@
-const { ApolloServer } = require('apollo-server');
+const { ApolloServer, AuthenticationError } = require('apollo-server');
 const typeDefs = require('./typeDefs');
 const resolvers = require('./resolvers');
 const connectDB = require('./config/db');
 
+const verifyUser = require('./utils/verifyUser');
+
 connectDB();
 require('dotenv').config();
 
 const server = new ApolloServer({
   typeDefs,
   resolvers,
+  context: ({ req }) => {
+    const { token } = req.headers || req.cookies || '';
+    return {
+      user: verifyUser(token),
```

Comment:
Please, refactor this method in a way that if a token is invalid or is absent it throws an exception here, not inside the service

Reply:
Please, also add a check by id that the user exists

Reply:
fixed